idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/Contents.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import org.immutables.value.Value;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Value.Immutable
+public abstract class Contents {
+    private static final Logger log = LoggerFactory.getLogger(Contents.class);
+
+    protected abstract String pageContent();
+
+    public static Optional<Contents> pageContents(URL pageUrl) {
```

Comment:
This class feels a bit odd being an immutables class. It does complicated mutable stuff (interact with intellij, http request) rather than just shuffling objects, then we extract the string out of it almost immediately. I'd consider just making this a static method on a utility class until we produce a type that has some behaviour (basically instance methods on it).

Reply:
So I think I've done what you where looking for here - I have converted this to a utility class and split the HTTP stuff into a separate method let me know if it looks good